examples/node/domstubs.js

```diff
@@ -91,24 +91,26 @@ DOMElement.prototype = {
   },
 
   toString: function DOMElement_toString() {
-    var attrList = [];
-    for (i in this.attributes) {
-      attrList.push(i + '="' + xmlEncode(this.attributes[i]) + '"');
+    var str = '<' + this.nodeName;
+    if (this.nodeName === 'svg:svg') {
+      str += ' xmlns:xlink="http://www.w3.org/1999/xlink"' +
+             ' xmlns:svg="http://www.w3.org/2000/svg"';
     }
+    for (var i in this.attributes) {
+      str += ' ' + i + '="' + xmlEncode(this.attributes[i]) + '"';
+    }
+
+    str += '>';
 
     if (this.nodeName === 'svg:tspan' || this.nodeName === 'svg:style') {
-      var encText = xmlEncode(this.textContent);
-      return '<' + this.nodeName + ' ' + attrList.join(' ') + '>' +
-             encText + '</' + this.nodeName + '>';
-    } else if (this.nodeName === 'svg:svg') {
-      var ns = 'xmlns:xlink="http://www.w3.org/1999/xlink" ' +
-               'xmlns:svg="http://www.w3.org/2000/svg"'
-      return '<' + this.nodeName + ' ' + ns + ' ' + attrList.join(' ') + '>' +
-             this.childNodes.join('') + '</' + this.nodeName + '>';
+      str += xmlEncode(this.textContent);
     } else {
-      return '<' + this.nodeName + ' ' + attrList.join(' ') + '>' +
```

Comment:
The issue is both in the attributes and the childNodes list. When I first changed `attrList` to `+=`, I was able to get 9 additional pages, whereas doing the full conversion gave only one more (on top of the 9). I haven't tested `childNodes` in isolation, but that is probably not too relevant.

A JS string in V8 can internally be represented in multiple ways. One of the representations is a sequence of substrings, which is especially useful in use cases involving concatenation, like this one. This string will be flattened at some point (e.g. when you read individual characters of the string).

Another reason for the current state of the patch (besides the boost in (peak) memory usage) is that it becomes easier to implement svg-to-file streaming (like a readable stream). It is conceptually as simple as replacing `+= extraData` with `.write(extraData)`.